x-pack/plugins/apm/server/lib/metrics/by_agent/default.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Setup } from '../../helpers/setup_request';
+import { getCPUChartData } from './shared/cpu';
+import { getMemoryChartData } from './shared/memory';
+import { PromiseReturnType } from '../../../../typings/common';
+
+export type DefaultMetricsChartsResponse = PromiseReturnType<
+  typeof getDefaultMetricsCharts
+>;
+
+export async function getDefaultMetricsCharts(
+  setup: Setup,
+  serviceName: string
+) {
```

Comment:
nit: some places we used destructed object as argument, other places like here positional arguments. Could we be more consistent in using one over the other; or is it a question of number of arguments (3 or more and it should be converted to an object kinda thing)?

Reply:
Yeah, to be honest I think I just evaluate each function API as I go. If it's one required arg and then an options hash, for instance, it may be `myFunc(myRequired, { optionalA = 'whatever' })` and when 2-3 are required you've convinced me at some point that TS makes it easy to just go with positional args haha. I'm happy to change this however you like, no strong feelings here. :)

Reply:
I'm fine with leaving it as-is and deciding on a case by case basis